### PR TITLE
En 1748 improvement of page's group handling

### PR DESCRIPTION
**Pull request by eugeniosant**

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
All is well

File: admin-console/pom.xml [hunk 1]
```diff
@@ -75,20 +75,6 @@
                     <localCheckout>true</localCheckout>
                 </configuration>
             </plugin>
```
Comment (joewhite101) [inline admin-console/pom.xml]: Why did you remove the sources jar?

File: admin-console/src/main/java/com/agiletec/apsadmin/portal/PageAction.java [hunk 2]
```diff
@@ -54,6 +54,18 @@ public void validate() {
         }
         this.checkCode();
         this.checkTitles();
+        if (this.getAuthorizationManager().isAuthOnGroup(this.getCurrentUser(), Group.ADMINS_GROUP_NAME)) {
+            try {
+                IPage currentPage = (this.getStrutsAction() == ApsAdminSystemConstants.EDIT) ? this.getUpdatedPage() : this.buildNewPage();
+                boolean check = this.getPageActionHelper().checkPageGroup(currentPage, this);
+                if (!check) {
+                    this.addActionError(this.getText("error.page.scanGroup"));
```
Comment (joewhite101) [inline admin-console/src/main/java/com/agiletec/apsadmin/portal/PageAction.java]: Log this at debug

File: engine/src/main/db/update/scriptPort.sql [hunk 3]
```diff
@@ -1,2 +1,5 @@
-INSERT INTO sysconfig (version, item, descr, config) VALUES ('production', 'dataTypeDefinitions', 'Definition of the Type Types', '<datatypes />');
```
Comment (joewhite101) [inline engine/src/main/db/update/scriptPort.sql]: This can be safely deleted? Does this have an impact on migration or upgrades?

File: engine/src/main/db/update/scriptServ.sql [hunk 4]
```diff
@@ -1,65 +1 @@
-CREATE TABLE dataobjects
```
Comment (joewhite101) [inline engine/src/main/db/update/scriptServ.sql]: Deleted on purpose?

File: engine/src/main/java/com/agiletec/aps/system/services/page/PageDAO.java [hunk 5]
```diff
@@ -1,896 +1,894 @@
...
+            }
+        }
+    }
+
+    protected PageRecord createPageRecord(String code, ResultSet res) throws Throwable {
+        PageRecord page = new PageRecord();
+        page.setCode(code);
+        page.setParentCode(res.getString(1));
+        page.setPosition(res.getInt(2));
+        if (res.getString(4) != null) {
```
Comment (joewhite101) [inline engine/src/main/java/com/agiletec/aps/system/services/page/PageDAO.java]: Double check this. I don't see a change in the number of columns selected in the query. The query changed but the column order looks the same?

File: engine/src/main/java/com/agiletec/aps/system/services/page/PageMetadata.java [hunk 6]
```diff
@@ -31,32 +32,44 @@
...
-     */
-    private PageModel model;
-
-    /**
-     * Toggle menu visibility on and off
-     */
-    private boolean showable = false;
-
-    private boolean useExtraTitles = false;
+    private static final Logger _logger = LoggerFactory.getLogger(PageMetadata.class);
```
Comment (joewhite101) [inline engine/src/main/java/com/agiletec/aps/system/services/page/PageMetadata.java]: New code with an underscore. Remove

File: engine/src/main/java/com/agiletec/aps/system/services/page/PageMetadata.java [hunk 7]
```diff
@@ -31,32 +32,44 @@
...
+    }
 
-    private Date updatedAt;
+    public void setGroup(String group) {
+        this._group = group;
+    }
 
     public ApsProperties getTitles() {
-        return titles;
+        return _titles;
```
Comment (joewhite101) [inline engine/src/main/java/com/agiletec/aps/system/services/page/PageMetadata.java]: resotored underscore

File: engine/src/main/java/com/agiletec/aps/system/services/page/PageMetadata.java [hunk 8]
```diff
@@ -66,20 +79,24 @@ public ApsProperties getTitles() {
      * codes of language.
      */
     public void setTitles(ApsProperties titles) {
-        this.titles = titles;
+        this._titles = titles;
```
Comment (joewhite101) [inline engine/src/main/java/com/agiletec/aps/system/services/page/PageMetadata.java]: Restored underscore

File: admin-console/pom.xml (hunk 1 above)
Comment (eugeniosant) [reply admin-console/pom.xml]: I inserted it in the main pom

File: admin-console/src/main/java/com/agiletec/apsadmin/portal/PageAction.java (hunk 2 above)
Comment (eugeniosant) [reply admin-console/src/main/java/com/agiletec/apsadmin/portal/PageAction.java]: Done

File: engine/src/main/java/com/agiletec/aps/system/services/page/PageDAO.java (hunk 5 above)
Comment (eugeniosant) [reply engine/src/main/java/com/agiletec/aps/system/services/page/PageDAO.java]: It's right. The fields changed are the following ones (from index 4)

File: engine/src/main/java/com/agiletec/aps/system/services/page/PageMetadata.java (hunk 6 above)
Comment (eugeniosant) [reply engine/src/main/java/com/agiletec/aps/system/services/page/PageMetadata.java]: removed

File: engine/src/main/java/com/agiletec/aps/system/services/page/PageMetadata.java (hunk 7 above)
Comment (eugeniosant) [reply engine/src/main/java/com/agiletec/aps/system/services/page/PageMetadata.java]: Removed

File: engine/src/main/java/com/agiletec/aps/system/services/page/PageMetadata.java (hunk 8 above)
Comment (eugeniosant) [reply engine/src/main/java/com/agiletec/aps/system/services/page/PageMetadata.java]: removed

**Comment by entando-service** [comment]
All is well
